apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -96,20 +104,49 @@ export default class JoinSection extends React.Component {
 
   handleKeyUp = event => {
     if (event.key === 'Enter') {
-      this.joinSection();
+      this.validateRecaptcha();
     } else if (event.key === 'Escape') {
-      this.setState(INITIAL_STATE);
+      this.setState({sectionCode: '', displayCaptcha: false});
     }
   };
 
-  joinSection = () => {
+  validateRecaptcha = () => {
+    const {sectionAttempts} = this.state;
+    const requireCaptcha = sectionAttempts >= 3;
+    if (requireCaptcha) {
+      this.setState({displayCaptcha: true});
+    } else {
+      this.joinSection();
+    }
+  };
+
+  close = () => {
+    //Remove the recaptcha script from DOM if necessary
+    const captchaScript = document.getElementById('captcha');
```

Comment:
Oh, is this part of the clean-up I referred to in the above comment?  It's a little confusing this to be asymmetric (with the creation in one class and the cleanup in another).

Reply:
I tried using `componentWillUnmount` to do some cleanup, but the way `BaseDialog` is written, the component itself doesn't unmount when you close the dialog. I do agree that the current way of doing this doesn't make much sense. I can create a `close` method in `ReCaptchaDialog` that first does the cleanup and then calls the `close` method passed in as `props`.

Reply:
I created a new method `onCloseCleanup` that cleans up the DOM and the `window` object after the Cancel or Submit buttons are pressed on the dialog.